### DISCUSSION TITLE
Set Thread numbers

### DISCUSSION THREAD

**Discussion by sneycampos** [php/frankenphp#1153] [Q&A] [answered]
Hello, i'm using FrankenPHP in multiple APIs in production but i'm trying to figure out how to set/have only 1 thread.

i have my Dockerfile and .env with num_threads 1
```Dockerfile
ENV FRANKENPHP_CONFIG="num_threads 1"
EXPOSE 80
```

```BASH
root@f3698a6ae797:/app# echo $FRANKENPHP_CONFIG
num_threads 1
```

But threre are some ?threads? and not only 1:
<img width="941" alt="image" src="https://github.com/user-attachments/assets/c626384d-893b-4da1-b862-980ca7fa424b">

Is this the expected behaviour or any tip to solve this?

**Comment by dunglas** [comment]
This setting configure the number of C threads used by PHP itself, but not the number of Go threads that are automatically managed by the Go runtime. I don't know if there is a way to configure the number of Go threads, but it's likely a bad idea. Similarly, having only 1 PHP thread will create a bottleneck. You should probably increase this number.

**Reply by sneycampos** [reply]
Thank you @dunglas

Was only a question but got it!

I am using this way only in a sandbox server running a few workers (laravel horizon) so my idea was reduce the memory usage because in sandbox we are using a low memory vps

**Comment by dunglas** [comment, marked as answer]
You should use the GOMEMLIMIT env var to do that, it works very well: https://frankenphp.dev/docs/performance/#go-runtime-configuration

**Reply by sneycampos** [reply]
@dunglas thank you! 
(not related, but: waiting the 1.3 php8.2 bookworm build 😄, did the build process miss something or was delayed?)

<img width="438" alt="image" src="https://github.com/user-attachments/assets/2491ca5e-66a4-4a1d-82da-680fdf924b47">
